QuickOSM/core/query_preparation.py

```diff
@@ -298,7 +299,7 @@ def prepare_query(self):
         self._query_is_ready = True
         return self._query_prepared
 
-    def prepare_url(self) -> str:
+    def prepare_url(self, convert: bool = False, output: str = "") -> str:
```

Comment:
Can you document acceptable values ? It seems `xml' and `mapql` ?

Maybe better to add a new definition in "definition/osm.py" about "QuerySyntax" or "QueryLanguage" and add "Xml" and "Oql". Then instead of giving **string** based object, you can give this new definition, which is a proper Python object.

Below
```python
if convert and output == QueryLanguage.Xml:
    query_string.addQueryItem('target','xml')
elif convert and output == QueryLanguage.Oql:
    query_string.addQueryItem('target','mapoql')
```

You can switch in this PR to use this new definition where needed.

Using enum objects avoids fragile code. String comparison may be fragile and it's good practice to good enum IMHO.

Doc about Python enum https://docs.python.org/3/library/enum.html#creating-an-enum